contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -3112,6 +3119,297 @@ exec_stmt_insert_bulk(PLtsql_execstate *estate, PLtsql_stmt_insert_bulk *stmt)
+		pfree(token);
+	}
+
+    db_name = get_cur_db_name();
+    if (dbcc_stmt.db_name)
+    {
+        if (pg_strcasecmp(db_name, dbcc_stmt.db_name) != 0)
+            is_cross_db = true;
+        db_name = pstrdup(dbcc_stmt.db_name);
+    }
+
+	if (!DbidIsValid(get_db_id(db_name)))
+	{
+		ereport(ERROR,
+		(errcode(ERRCODE_UNDEFINED_DATABASE),
+			errmsg("database \"%s\" does not exist", db_name)));
+		pfree(db_name);
+	}
+
+	login_is_db_owner = 0 == strncmp(GetUserNameFromId(GetSessionUserId(), false),
+										get_owner_of_db(db_name), NAMEDATALEN);
+	user = get_user_for_database(db_name);
+	guest_role_name = get_guest_role_name(db_name);
+
+    if(is_cross_db)
+    {
+        if (user)
+            SetCurrentRoleId(GetSessionUserId(), false);
+        else
+            ereport(ERROR,
+                    (errcode(ERRCODE_UNDEFINED_DATABASE),
+                     errmsg("The server principal \"%s\" is not able to access "
+                            "the database \"%s\" under the current security context",
+                            login, dbcc_stmt.db_name)));
+    }
+    pfree(login);
+
+	if (dbcc_stmt.schema_name)
+	{
+		schema_name = pstrdup(dbcc_stmt.schema_name);
+		nsp_name = get_physical_schema_name(db_name, schema_name);
+	}
+	else
+	{
+		/* 
+		 * If schema_name is not provided, find default schema for current user
+		 * and get physical schema name
+		 */
+		if (!user)
+		{
+			ereport(ERROR,
+			(errcode(ERRCODE_UNDEFINED_OBJECT),
+				 errmsg("user does not exist")));
+		}
+		else if ((guest_role_name && strcmp(user, guest_role_name) == 0))
+		{
+			nsp_name = pstrdup(get_guest_schema_name(db_name));
+		}
+		else
+		{
+			schema_name = get_authid_user_ext_schema_name((const char *) db_name, user);
+			nsp_name = get_physical_schema_name(db_name, schema_name);
+		}
+	}
+
+	pfree(db_name);
+
+	nsp_oid = get_namespace_oid(nsp_name, false);
+	if(!OidIsValid(nsp_oid))
+	{
+		ereport(ERROR,
+		(errcode(ERRCODE_UNDEFINED_SCHEMA),
+			errmsg("schema \"%s\" does not exist", schema_name)));
+	}
+
+	/* Permission check */
+	if (!(pg_namespace_ownercheck(nsp_oid, GetUserId()) ||
+			has_privs_of_role(GetSessionUserId(), get_role_oid("sysadmin", false)) ||
+				login_is_db_owner))
+		aclcheck_error(ACLCHECK_NOT_OWNER, OBJECT_SCHEMA, nsp_name);
+
+	table_name = pstrdup(dbcc_stmt.table_name);
+	table_oid = get_relname_relid(table_name, nsp_oid);
+	if(!OidIsValid(table_oid))
+	{
+		ereport(ERROR,
+		(errcode(ERRCODE_UNDEFINED_TABLE),
+			errmsg("relation \"%s\" does not exist", table_name)));
+	}
+
+	rel = RelationIdGetRelation(table_oid);
+	tupdesc = RelationGetDescr(rel);
+
+	/* Find Identity column in table and associated sequence */
+	for (attnum = 0; attnum < tupdesc->natts; attnum++)
+	{
+		Form_pg_attribute attr = TupleDescAttr(tupdesc, attnum);
+
+		if (attr->attidentity)
+		{
+			attname = NameStr(attr->attname);
+			seqid = getIdentitySequence(table_oid, attnum + 1, false);
+			break;
+		}
+	}
+
+	RelationClose(rel);
```

Comment:
We already have helper function for this. Let's use the same - get_table_identity().

Reply:
Although we can use the `get_table_identity()` api to find `seqid` but we will also need the name of this sequence  col  that is `attname` in the code  to find maximum value. 

here - https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1646/files#diff-3ca2886287e484395b98c3d6f90cb766e4f75c5cd9ef1e975c1bb1c79b93e7ab:~:text=query%20%3D%20psprintf,%2C%20table_name)%3B